eventuals/lock.h

```diff
@@ -694,20 +694,41 @@ class ConditionVariable {
     return eventuals::Wait(
         lock_,
         [this, f = std::move(f), waiter = Waiter()](auto notify) mutable {
-          waiter.notify = std::move(notify);
-          if (head_ == nullptr) {
-            head_ = &waiter;
-          } else if (head_->next == nullptr) {
-            head_->next = &waiter;
+          // Check if template `F` is for `EmptyCondition`.
+          constexpr bool using_empty_condition = std::is_same_v<F, EmptyCondition>;
+
+          // Determine current state of condition variable.
+          // Note: Initializing using an `?:` did not work, so here goes verbatim.
+          bool current_condition;
+          if constexpr (using_empty_condition) {
+            current_condition = f(waiter);
           } else {
-            auto* next = head_->next;
-            while (next->next != nullptr) {
-              next = next->next;
+            current_condition = f();
+          }
+
+          if (current_condition == true) {
```

Comment:
Based on my comment in the code above (which I wasn't able to do correctly because GitHub won't let me make a suggestion on deleted lines 🤦🏻‍♂️), then this becomes:
```suggestion
          if (wait()) {
```